src/d3a/setup/two_sided_pay_as_clear/test_clearing_energy.py

```diff
@@ -0,0 +1,71 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+from d3a.models.appliance.simple import SimpleAppliance
+from d3a.models.appliance.switchable import SwitchableAppliance
+from d3a.models.area import Area
+from d3a.models.strategy.commercial_producer import CommercialStrategy
+from d3a.models.strategy.load_hours import LoadHoursStrategy
+from d3a.models.appliance.pv import PVAppliance
+from d3a.models.strategy.pv import PVStrategy
+from d3a.models.strategy.storage import StorageStrategy
+from d3a.models.const import ConstSettings
+
+
+def get_setup(config):
+
+    ConstSettings.IAASettings.MARKET_TYPE = 3
+    ConstSettings.GeneralSettings.MARKET_CLEARING_FREQUENCY_PER_SLOT = 3
+    ConstSettings.LoadSettings.INITIAL_BUYING_RATE = 35
+    ConstSettings.LoadSettings.FINAL_BUYING_RATE = 35
+    ConstSettings.StorageSettings.MIN_BUYING_RATE = 24.99
+    ConstSettings.StorageSettings.BREAK_EVEN_BUY = 25
+    ConstSettings.StorageSettings.BREAK_EVEN_SELL = 25.01
+
+    area = Area(
+        'Grid',
+        [
+            Area(
```

Comment:
Well, this is ok for now, but the problem with this file is that it does not test offer/bid forwarding, because there is only one house. If there were 2 houses we would be able to see whether bids/offers are forwarded properly. But anyways, as I said before, it is ok for now

Reply:
Well, the target of this setup file is-to specifically test if cumulative traded offered_energy equals to bid_energy within an area. Not sure if this is the right approach.